src/pandablocks_ioc/ioc.py

```diff
@@ -863,9 +867,35 @@ def _make_pos_out(
         )
 
         capture_record_name = EpicsName(record_name + ":CAPTURE")
+        dataset_record_name = EpicsName(record_name + ":DATASET")
         labels, capture_index = self._process_labels(
             field_info.capture_labels, values[capture_record_name]
         )
+
+        def adjust_hdf_field_name_based_on_dataset(dataset_name) -> str:
+            current_capture_mode = labels[record_dict[capture_record_name].record.get()]
+
+            # Throw away all the old settings
+            self._capture_record_hdf_names[record_name] = {}
+
+            # Add a -Max or -Min or -Mean to the dataset name if it exists
+            if current_capture_mode in ("Min Max", "Min Max Mean"):
+                for capture_mode_instance in current_capture_mode.split(" "):
+                    self._capture_record_hdf_names[record_name][
+                        capture_mode_instance
+                    ] = f"{(dataset_name or record_name)}-{capture_mode_instance}"
+
+            # Current capture mode will be the same as the capture mode instance
+            else:
+                self._capture_record_hdf_names[record_name][current_capture_mode] = (
+                    dataset_name or f"{record_name}-{current_capture_mode}"
+                )
```

Comment:
Please can we make sure that suffixing with `-min` and `-max` is the special case? Also, I would be happy if the dataset table only contained those that explicitly contained fields with named datasets:

```suggestion
            field_name = record_name.replace(":", ".")
            # Throw away all the old settings
            self._capture_record_hdf_names[field_name] = capture_names = {}
            # Only consider explicitly named datsets
            if dataset_name:
                # The last capture option is the one that should take the dataset name
                if current_capture_mode != "No":
                    capture_names[current_capture_mode.split(" ")[-1]] = dataset_name
                # But also suffix -min and -max if both are present
                if "Min Max" in current_capture_mode:
                    capture_names["Min"] = f"{dataset_name}-min"
                    capture_names["Max"] = f"{dataset_name}-min"
```

Reply:
Yep, this is much nicer.